presto-main/src/main/java/com/facebook/presto/operator/GroupIdOperator.java

```diff
@@ -44,22 +46,24 @@
         private final List<Type> inputTypes;
         private final List<Type> outputTypes;
         private final List<List<Integer>> groupingSetChannels;
+        private final Map<Integer, Integer> passthroughChannelMapping;
 
         private boolean closed;
 
         public GroupIdOperatorFactory(
                 int operatorId,
                 PlanNodeId planNodeId,
                 List<? extends Type> inputTypes,
-                List<List<Integer>> groupingSetChannels)
+                List<? extends Type> outputTypes,
+                List<List<Integer>> groupingSetChannels,
+                Map<Integer, Integer> passthroughChannelMapping)
```

Comment:
Because we also need to know what is the channel number they need to be copied to.